lib/src/markdown_processor.dart

```diff
@@ -181,12 +184,73 @@ final List<md.BlockSyntax> _markdownBlockSyntaxes = [
 final RegExp _hideSchemes = RegExp('^(http|https)://');
 
 class MatchingLinkResult {
-  final ModelElement element;
+  final ModelElement modelElement;
   final bool warn;
 
-  MatchingLinkResult(this.element, {this.warn = true});
+  MatchingLinkResult(this.modelElement, {this.warn = true});
+
+  @override
+  bool operator ==(Object other) {
+    return other is MatchingLinkResult &&
+        modelElement == other.modelElement &&
+        warn == other.warn;
+  }
+
+  @override
+  int get hashCode => hash2(modelElement, warn);
+
+  bool isEquivalentTo(MatchingLinkResult other) {
+    if (this == other) return true;
+    if (modelElement?.canonicalModelElement ==
+        other.modelElement?.canonicalModelElement) return true;
+    if (other.modelElement is Parameter && modelElement == null) {
```

Comment:
Would you mind adding comment(s) for these two cases? I'm not sure I understand why they're 'equivalent'.